packages/events/src/router/middleware/validate/index.ts

```diff
@@ -303,74 +319,77 @@ function validateCorrectableFields({
   return errors
 }
 
-export function validateAction(actionType: ActionType) {
-  const fn: MiddlewareFunction<
-    TrpcContext,
-    OpenApiMeta,
-    TrpcContext,
-    TrpcContext,
-    ActionInputWithType
-  > = async ({ input, ctx, next }) => {
-    const event = await getEventById(input.eventId)
-    const eventConfig = await getEventConfigurationById({
-      token: ctx.token,
-      eventType: event.type
-    })
+export const validateAction: MiddlewareFunction<
+  TrpcContext,
+  OpenApiMeta,
+  TrpcContext,
+  TrpcContext & { isDuplicateAction?: boolean; event: EventDocument },
+  ActionInputWithType
+> = async ({ input, next, ctx }) => {
+  const actionType = input.type
```

Comment:
`validateAction` doesn't need to be a function, rather it can be a middleware itself, since `actionType` can already be accessed from `input`